pipeline/vars/common.groovy

```diff
@@ -0,0 +1,127 @@
+            sudo mount -t nfs -o sec=sys,nfsvers=4.1 reesi004.ceph.redhat.com:/ /ceph
+        fi
+
+        # Copy the auth files from internal server
+        wget http://magna002.ceph.redhat.com/cephci-jenkins/.osp-cred-ci-2.yaml -O ${HOME}/osp-cred-ci-2.yaml
+        wget http://magna002.ceph.redhat.com/cephci-jenkins/.cephci.yaml -O ${HOME}/.cephci.yaml
+
+        # Install cephci pre-requisistes
+        rm -rf .venv
+        python3 -m venv .venv
+        source .venv/bin/activate
+        python -m pip install --upgrade pip
+        python -m pip install -r requirements.txt
+        deactivate
+    '''
+}
+
+def runTestSuite() {
+    /*
+        Execute the test suite by processing CI_MESSAGE for container
+        information along with the environment variables. The required
+        variables are
+            - sutVMConf
+            - sutConf
+            - testSuite
+            - addnArgs
+    */
+    println "Begin Test Suite execution"
+
+    // generate random instance name
+    def sec = sh(script: "echo \$(date +%s)", returnStdout: true).trim()
+    instanceName = "psi-${sec}"
+
+    def rhceph_ver = "${env.rhceph_ver}" ?: '5.0-rhel-8'
+
+    // Build the CLI options
+    def cmd = "${env.WORKSPACE}/.venv/bin/python run.py"
+    cmd += " --osp-cred ${env.HOME}/osp-cred-ci-2.yaml"
+    cmd += " --instances-name ${instanceName}"
+
+    // Append test suite specific
+    cmd += " --inventory ${env.sutVMConf}"
+    cmd += " --global-conf ${env.sutConf}"
+    cmd += " --suite ${env.testSuite}"
+
+    // If CI_MESSAGE is populated, the preference is use the information.
+    def ciMessage = "${params.CI_MESSAGE}" ?: ""
+    if (ciMessage?.trim()) {
+        // Process the CI Message
+        def jsonParser = new JsonSlurper()
+        def jsonCIMsg = jsonParser.parseText("${params.CI_MESSAGE}")
+
+        composeId = jsonCIMsg.compose_id
+        def composeUrl = jsonCIMsg.compose_url
+
+        def (dockerDTR, dockerImage1, dockerImage2Tag) = (jsonCIMsg.repository).split('/')
+        def (dockerImage2, dockerTag) = dockerImage2Tag.split(':')
+        def dockerImage = "${dockerImage1}/${dockerImage2}"
+
+        // get rhbuild value from RHCEPH-5.0-RHEL-8.yyyymmdd.ci.x
+        def composeStrings = composeId.split('-')
+        rhceph_ver = composeId.substring(7,16).toLowerCase()
+
+        cmd += " --docker-registry ${dockerDTR}"
+        cmd += " --docker-image ${dockerImage}"
+        cmd += " --docker-tag ${dockerTag}"
+
+        cmd += " --rhbuild ${rhceph_ver}"
+        cmd += " --rhs-ceph-repo ${composeUrl}"
+        cmd += " --insecure-registry"
+        cmd += " --ignore-latest-container"
+
+    } else {
+        cmd += " --rhbuild ${rhceph_ver}"
+    }
+
+    // Check for additional arguments
+    def addnArgFlag = "${env.addnArgs}" ?: ""
+    if (addnArgFlag?.trim()) {
+        cmd += " ${env.addnArgs}"
+    }
+
+    //command execution
+    rc = sh(script: "PYTHONUNBUFFERED=1 ${cmd}", returnStatus: true)
```

Comment:
Do we need to check exit status? Since on test failure jenkins stage will be failed and exited.

Reply:
Yes, that is correct. I have made the required changes.